### PR TITLE
feat: Spark Operator Blueprint update

### PR DESCRIPTION
**Pull request by alanty**
### What does this PR do?

This PR includes updates/bumps and changes that I've made while working with the Spark Operator Blueprint:

- Update kubernetes version to 1.28 (current EKS latest)
- Update `providers.tf` to use `exec`for authentication tokens
- Fix typo `aws-cloudwatch-metrics-valyes.yaml`
- Add `AmazonSSMManagedInstanceCore` IAM policy to Karpenter node role
- Remove `kubecost` scrape, add `karpenter` config
- Decrease scrape_interval for Yunikorn metrics 1m -> 15s
- Set Yunikorn `nodesortpolicy` to `binpacking` 
- remove unused `aws_eks_cluster_auth data` resource
- update Karpenter Provisioners to leverage `local-disks` in bootstrap.sh
- remove `hostpath` mounts from NVMe examples
- update tcpds benchmark, increase Yunikorn prod queue max to accommodate job.
- add queue labels to pyspark examples

### Motivation
Been working with the example and I've updated and fixed a few things that may help someone else using the blueprint.

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes

- I removed the scrape config for kubecost metrics because there were duplicate metrics for things like kube-state-metrics that were breaking queries in Grafana dashboards.
- I adjusted the scrape_interval for Yunikorn because 1m wasn't frequent enough for their dashboards to calculate data
- Since we're using the [`local-disks` script from the EKS AMI](https://github.com/awslabs/amazon-eks-ami/pull/1171) the pods will write to an NVMe raid by default, we don't need the hostpath mounts to leverage the fast disks.

### CODE REVIEW

File: analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-3t.yaml [hunk 1]
```diff
@@ -65,22 +69,7 @@ spec:
...
-        mountPath: /ossdata1
-        readOnly: false
-    initContainers:
-    - name: volume-permission
-      image: public.ecr.aws/y4g4v0z7/busybox
-      command: ['sh', '-c', 'mkdir /ossdata1; chown -R 1000:1000 /ossdata1']
-      volumeMounts:
-        - name: spark-local-dir-1
-          mountPath: /ossdata1
     cores: 4
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-3t.yaml:73]: We can add a comment stating that the NVMe SSD disks provided with c5d instances are automatically formatted and mounted by Karpenter, and subsequently integrated into the node as the primary storage volume. Therefore, it is unnecessary to employ hostPath in Spark jobs for mounting this volume to pods. Instead, Spark pods can effortlessly utilize the local NVMe SSD through the use of emptyDir().

File: analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-3t.yaml [hunk 2]
```diff
@@ -9,6 +9,10 @@ kind: SparkApplication
 metadata:
   name: tpcds-benchmark-3tb
   namespace: spark-team-a
+  labels:
+    app: "tpcds-benchmark"
+    applicationId: "tpcds-benchmark-3t"
+    queue: root.prod
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-3t.yaml:15]: add a comment here saying  "YuniKorn Queue"

File: analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-data-generation-3t.yaml [hunk 3]
```diff
@@ -8,6 +8,10 @@ kind: SparkApplication
 metadata:
   name: tpcds-data-generation-3t
   namespace: spark-team-a
+  labels:
+    app: "tpcds-data-generation"
+    applicationId: "tpcds-data-generation-3t"
+    queue: root.prod
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-data-generation-3t.yaml:14]: same as above

File: analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-data-generation-3t.yaml [hunk 4]
```diff
@@ -64,22 +68,7 @@ spec:
...
-      - name: spark-local-dir-1
-        mountPath: /ossdata1
-        readOnly: false
-    initContainers:
-    - name: volume-permission
-      image: public.ecr.aws/y4g4v0z7/busybox
-      command: ['sh', '-c', 'mkdir /ossdata1; chown -R 1000:1000 /ossdata1']
-      volumeMounts:
-        - name: spark-local-dir-1
-          mountPath: /ossdata1
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/benchmark/tpcds-benchmark-data-generation-3t.yaml:82]: We can add a comment stating that the NVMe SSD disks provided with c5d instances are automatically formatted and mounted by Karpenter, and subsequently integrated into the node as the primary storage volume. Therefore, it is unnecessary to employ hostPath in Spark jobs for mounting this volume to pods. Instead, Spark pods can effortlessly utilize the local NVMe SSD through the use of emptyDir().

File: analytics/terraform/spark-k8s-operator/examples/cluster-autoscaler/nvme-ephemeral-storage/nvme-ephemeral-storage.yaml [hunk 5]
```diff
@@ -79,24 +79,7 @@ spec:
...
-      - name: spark-local-dir-1
-        mountPath: /data1
-        readOnly: false
-    initContainers:
-      - name: volume-permissions
-        image: public.ecr.aws/y4g4v0z7/busybox
-        command: [ 'sh', '-c', 'chown -R 185 /local1' ]
-        volumeMounts:
-          - mountPath: "/local1"
-            name: "spark-local-dir-1"
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/cluster-autoscaler/nvme-ephemeral-storage/nvme-ephemeral-storage.yaml:99]: We can add a comment stating that the NVMe SSD disks provided with c5d instances are automatically formatted and mounted by Karpenter, and subsequently integrated into the node as the primary storage volume. Therefore, it is unnecessary to employ hostPath in Spark jobs for mounting this volume to pods. Instead, Spark pods can effortlessly utilize the local NVMe SSD through the use of emptyDir().

File: analytics/terraform/spark-k8s-operator/examples/cluster-autoscaler/nvme-yunikorn-gang-scheduling/nvme-storage-yunikorn-gang-scheduling.yaml [hunk 6]
```diff
@@ -79,24 +79,7 @@ spec:
...
-      - name: spark-local-dir-1
-        mountPath: /data1
-        readOnly: false
-    initContainers:
-      - name: volume-permissions
-        image: public.ecr.aws/y4g4v0z7/busybox
-        command: [ 'sh', '-c', 'chown -R 185 /local1' ]
-        volumeMounts:
-          - mountPath: "/local1"
-            name: "spark-local-dir-1"
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/cluster-autoscaler/nvme-yunikorn-gang-scheduling/nvme-storage-yunikorn-gang-scheduling.yaml:99]: We can add a comment stating that the NVMe SSD disks provided with c5d instances are automatically formatted and mounted by Karpenter/CA, and subsequently integrated into the node as the primary storage volume. Therefore, it is unnecessary to employ hostPath in Spark jobs for mounting this volume to pods. Instead, Spark pods can effortlessly utilize the local NVMe SSD through the use of emptyDir().

File: analytics/terraform/spark-k8s-operator/examples/karpenter/nvme-ephemeral-storage/nvme-ephemeral-storage.yaml [hunk 7]
```diff
@@ -79,24 +79,7 @@ spec:
...
-      - name: spark-local-dir-1
-        mountPath: /data1
-        readOnly: false
-    initContainers:
-      - name: volume-permissions
-        image: public.ecr.aws/y4g4v0z7/busybox
-        command: [ 'sh', '-c', 'chown -R 185 /mnt/k8s-disks' ]
-        volumeMounts:
-          - mountPath: "/mnt/k8s-disks"
-            name: "spark-local-dir-1"
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/karpenter/nvme-ephemeral-storage/nvme-ephemeral-storage.yaml:99]: We can add a comment stating that the NVMe SSD disks provided with c5d instances are automatically formatted and mounted by Karpenter, and subsequently integrated into the node as the primary storage volume. Therefore, it is unnecessary to employ hostPath in Spark jobs for mounting this volume to pods. Instead, Spark pods can effortlessly utilize the local NVMe SSD through the use of emptyDir().

File: analytics/terraform/spark-k8s-operator/examples/karpenter/nvme-yunikorn-gang-scheduling/nvme-storage-yunikorn-gang-scheduling.yaml [hunk 8]
```diff
@@ -79,24 +79,8 @@ spec:
...
-      - name: spark-local-dir-1
-        mountPath: /data1
-        readOnly: false
-    initContainers:
-      - name: volume-permissions
-        image: public.ecr.aws/y4g4v0z7/busybox
-        command: [ 'sh', '-c', 'chown -R 185 /mnt/k8s-disks' ]
-        volumeMounts:
-          - mountPath: "/mnt/k8s-disks"
-            name: "spark-local-dir-1"
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/examples/karpenter/nvme-yunikorn-gang-scheduling/nvme-storage-yunikorn-gang-scheduling.yaml:99]: We can add a comment stating that the NVMe SSD disks provided with c5d instances are automatically formatted and mounted by Karpenter, and subsequently integrated into the node as the primary storage volume. Therefore, it is unnecessary to employ hostPath in Spark jobs for mounting this volume to pods. Instead, Spark pods can effortlessly utilize the local NVMe SSD through the use of emptyDir().

File: analytics/terraform/spark-k8s-operator/helm-values/kube-prometheus-amp-enable.yaml [hunk 9]
```diff
@@ -28,22 +28,11 @@ prometheus:
...
-        honor_labels: true
-        scrape_interval: 1m
-        scrape_timeout: 10s
-        metrics_path: /metrics
-        scheme: http
-        dns_sd_configs:
-          - names:
-              - kubecost-cost-analyzer.kubecost.svc
-            type: 'A'
-            port: 9003
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/helm-values/kube-prometheus-amp-enable.yaml:43]: Are you not using KubeCost?

File: analytics/terraform/spark-k8s-operator/helm-values/yunikorn-values.yaml [hunk 10]
```diff
@@ -107,6 +107,8 @@ yunikornDefaults:
   queues.yaml: |
     partitions:
       - name: default
+        nodesortpolicy:
+          type: binpacking
```
Comment (vara-bonthu) [inline analytics/terraform/spark-k8s-operator/helm-values/yunikorn-values.yaml:111]: This is very important feature: Add some details to explain this feature

**Review by vara-bonthu — COMMENTED** [review]
@alanty, I appreciate your PR! I've added a few minor suggestions for your review. Additionally, could you please ensure that any necessary updates are reflected in the Website documentation? You can refer to the TPCDS Benchmark test example at this link: https://awslabs.github.io/data-on-eks/docs/blueprints/data-analytics/spark-operator-yunikorn, particularly in the example section. It would also be beneficial to provide an explanation on the utilization of NVMe SSD by the pods in this configuration.

File: analytics/terraform/spark-k8s-operator/helm-values/kube-prometheus-amp-enable.yaml (hunk 9 above)
Comment (alanty) [reply analytics/terraform/spark-k8s-operator/helm-values/kube-prometheus-amp-enable.yaml:43]: Kubecost is still running but scraping these metrics was causing some problems with grafana dashboards. Queries were returning multiple metrics and breaking "group by" statements. 

Kubecost should still be able to collect metrics and make determinations on cost, it just isn't scraped in the central Prometheus config. 
happy to revert these changes as needed, i didn't investigate the issue very far and don't want to break other stuff.

File: analytics/terraform/spark-k8s-operator/helm-values/yunikorn-values.yaml (hunk 10 above)
Comment (alanty) [reply analytics/terraform/spark-k8s-operator/helm-values/yunikorn-values.yaml:112]: added a quick call out in the values file and link to the policy docs. 

I didn't find a great place on the site to add a callout, maybe we should add a section on Yunikorn and some of the benefits/config we have?

**Review by vara-bonthu — APPROVED** [review]
LGTM 👍🏼 Thanks for the PR @alanty 🔥

File: analytics/terraform/spark-k8s-operator/helm-values/kube-prometheus-amp-enable.yaml [hunk 11]
```diff
@@ -52,6 +41,16 @@ prometheus:
...
+      - job_name: karpenter
+        kubernetes_sd_configs:
+        - role: endpoints
+          namespaces:
+            names:
+            - karpenter
+        relabel_configs:
+        - source_labels: [__meta_kubernetes_endpoint_port_name]
+          regex: http-metrics
+          action: keep
```
Comment (askulkarni2) [inline analytics/terraform/spark-k8s-operator/helm-values/kube-prometheus-amp-enable.yaml:53]: Nice!

File: analytics/terraform/spark-k8s-operator/variables.tf [hunk 12]
```diff
@@ -12,7 +12,7 @@ variable "region" {
 
 variable "eks_cluster_version" {
   description = "EKS Cluster version"
-  default     = "1.26"
+  default     = "1.28"
```
Comment (askulkarni2) [inline analytics/terraform/spark-k8s-operator/variables.tf:15]: 🙌🏽

**Review by askulkarni2 — APPROVED** [review]
@alanty LGTM! Please merge latest main so CI checks can pass.

**Pull request merged**